test/framework/flux.go

```diff
@@ -376,6 +376,9 @@ func (e *ClusterE2ETest) PushWorkloadClusterToGit(w *WorkloadCluster, opts ...ap
 
 	// Update the cluster config with the provided api.ClusterConfigFillers
 	w.UpdateClusterConfig(opts...)
+	if w.clusterStateConfig != nil {
+		w.updateClusterStateConfig()
```

Comment:
why is this needed?

Reply:
I was going for keeping the spec in the `clusterValidatorConfig` up to date with the in the running cluster when the API is used.